### ISSUE TITLE
IOError: Unable to read attribute (no appropriate function for conversion path)

### ISSUE THREAD

**Issue by WxmanJ** [pytroll/satpy#233]
Encountering an error using the abi_l1b reader. 

```
[DEBUG:` 2018-03-19 11:29:11 : satpy.readers.yaml_reader] Assigning to abi_l1b: ['/root/goesr_data/OR_ABI-L1b-RadC-M3C02_G16_s20180781507251_e20180781510024_c20180781510067.nc']
Traceback (most recent call last):
  File "goesr.py", line 17, in <module>
    global_scene = Scene(reader='abi_l1b',filenames=FILES)
  File "/usr/lib/python2.7/site-packages/satpy/scene.py", line 143, in __init__
    reader_kwargs=reader_kwargs)
  File "/usr/lib/python2.7/site-packages/satpy/scene.py", line 188, in create_reader_instances
    ppp_config_dir=self.ppp_config_dir)
  File "/usr/lib/python2.7/site-packages/satpy/readers/__init__.py", line 412, in load_readers
    reader_instance.create_filehandlers(loadables)
  File "/usr/lib/python2.7/site-packages/satpy/readers/yaml_reader.py", line 599, in create_filehandlers
    filename_set)
  File "/usr/lib/python2.7/site-packages/satpy/readers/yaml_reader.py", line 588, in new_filehandlers_for_filetype
    return list(filtered_iter)
  File "/usr/lib/python2.7/site-packages/satpy/readers/yaml_reader.py", line 562, in filter_fh_by_metadata
    filehandler.metadata['start_time'] = filehandler.start_time
  File "/usr/lib/python2.7/site-packages/satpy/readers/abi_l1b.py", line 172, in start_time
    return datetime.strptime(self.nc.attrs['time_coverage_start'].decode(), '%Y-%m-%dT%H:%M:%S.%fZ')
  File "/usr/lib/python2.7/site-packages/h5netcdf/attrs.py", line 19, in __getitem__
    return self._h5attrs[key]
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "/usr/lib64/python2.7/site-packages/h5py/_hl/attrs.py", line 81, in __getitem__
    attr.read(arr, mtype=htype)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5a.pyx", line 355, in h5py.h5a.AttrID.read
  File "h5py/_proxy.pyx", line 36, in h5py._proxy.attr_rw
IOError: Unable to read attribute (no appropriate function for conversion path)
```

After a little research on forums, I found a similar error mentioning (below). Not sure if this is relevant, but I wanted to add it as a note.

``` 
The issue here is that recent versions of netCDF-C save the NC_CHAR dtype as fixed length UTF8 strings, which h5py cannot read. 
```

**Comment by davidh-ssec** [comment]
@WxmanJ Please use 3 back ticks (`) before and after all code/terminal output.

I believe this is the same issue Ron Goodson had found (can't find his github username right now). This is because the default netcdf library used in older versions of satpy is `h5netcdf`. In future versions we will use the default from xarray which is typically netcdf4-python. @WxmanJ if you are able to you could also try updating your version of the HDF5 library and reinstalled/updating the version of h5py you are using. It looks like you are using system libraries, I suggest using conda if at all possible so you can get newer versions of packages.

**Comment by WxmanJ** [comment]
@davidh-ssec I did use 3 back ticks before and after the debug code. Is there something else you are referring to?

Would I update the HDF5 library and h5py using pip? I am looking at migrating to conda. Thank you for the tip. Also, do you know when the next official release will be out or is there just a discrepancy of what packages are available on pip versus conda?

**Comment by davidh-ssec** [comment]
Make sure the backticks are on their own separate lines. So 3 ticks on one line, next one or more lines put your text, next line 3 backticks.

HDF5 in your setup is from the system package manager and may not have newer versions available. H5py could be from the system manager or pip depending on how you installed it.

**Comment by davidh-ssec** [comment]
@mraspaud is making an alpha release today I believe. I'm heading back from a trip so I don't know all the details. Regardless this alpha version of 0.9 has a lot of changes and has not been tested using the SCMI writer.

**Comment by davidh-ssec** [comment]
FYI: https://github.com/h5py/h5py/issues/719

**Comment by mraspaud** [comment]
What's the status on this one @djhoese @WxmanJ  ?

**Comment by djhoese** [comment]
@mraspaud I think so. The newest 0.9 alpha release uses xarray's default NetCDF engine (netcdf4-python) to read the files so this shouldn't be an issue anymore (and was probably fixed a couple alpha releases ago). This will not be fixed in 0.8.x.
